src/tests/ftest/get_remote_files.sh

```diff
@@ -0,0 +1,369 @@
+#    without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+# THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#######################################
+# Print usage to user.
+#######################################
+usage() {
+    echo -n "get_remote_files.sh [OPTION]...
+
+This script has 4 modes that can be executed independently if needed.
+
+ 1. Display big files specified by threshold value i.e. -t 5mb. This option
+    will list files within provided local path with -f option and will prepend
+    each file with a \"Y\" if file exceeds threshold value and with \"N\" if
+    it does not.
+
+ 2. Compress files specified in provided local path with -f option that exceed
+    1M in size. This script will check if current local system is a VM host, if
+    so, compression will not be executed.
+
+ 3. Archive files specified in provided local path with -f option to provided
+    remote path destination.
+
+ 4. Create and display cart_logtest log files created by running cart_logtest.py
+    script. Output of running the script will be redirected to *_ctestlog.log
+    files.
+
+ Options:
+      -z        Compress files
+      -t        Threshold value to determine classification of big logs
+      -a        Remote path to scp files to i.e. server-A:/path/to/file
+      -f        Local path to files for archiving/compressing/size checking
+      -c        Enable running cart_logtest.py
+      -x        Enable performing compression when running on VM system
+      -v        Display commands being executed
+      -h        Display this help and exit
+"
+}
+
+#######################################
+# CONSTANT variables.
+#######################################
+# shellcheck disable=SC2046,SC2086
+CART_LOGTEST_PATH="$(dirname $(readlink -f ${0}))/cart/cart_logtest.py"
+
+#######################################
+# Check if the files exist on this host.
+# Arguments:
+#   $1: local logs to process
+# Returns:
+#   0: Files exist or no matches were found
+#   1: Missing the log files argument
+#######################################
+check_files_input() {
+    if [ -n "${1}" ]; then
+        # shellcheck disable=SC2086
+        files=$(ls -d ${1} 2> /dev/null)
+        files_rc=$?
+        if [ -n "${files}" ] && [ "${files_rc}" -eq 0 ]; then
```

Comment:
This won't work with `set -e` enabled as the script will exit if `ls` exits `!0`:

```suggestion
        if ! files=$(ls -d ${1} 2> /dev/null); then
```

Reply:
Will fix